fixed headshots.py creating folders and clarifed variables

Jdavies08 opened the pull request.

Jdavies08 committed:
fixed headshots.py creating folders and clarifed variables

Jdavies08 commented:
two errors name need replacing with NAME and the second folder needs to make dataset/name not /name

Jdavies08 committed:
Fixed error with directory not being made correctly

Jdavies08 committed:
added support for multiple faces and unknowns

now multiple faces can be detected and unknowns will be labeled as well as
now handling the absence of an encodings file

Jdavies08 committed:
clarified to match pylint standards

HarrisonHG commented on headshots.py:
```diff
@@ -1,19 +1,26 @@
 
 DATASET_FOLDER = "dataset"
-NAME = 'donkeykong' #replace with your name
+NAME = 'user2' #replace with your name
```
If it's going to be needed at every usage, perhaps it would be a good idea to turn this into a normal variable (lower case) and ask input from the user on the console.

HarrisonHG commented on headshots.py:
```diff
@@ -1,19 +1,26 @@
+    os.makedirs(DATASET_FOLDER +"/"+ NAME)
 
 cv2.namedWindow("press space to take a photo", cv2.WINDOW_NORMAL)
 cv2.resizeWindow("press space to take a photo", 500, 300)
```
move 500 and 300 to WINDOW_WIDTH and WINDOW_HEIGHT constants

HarrisonHG commented on main.py:
```diff
@@ -12,9 +13,13 @@
-with open(SAVE_FILE, "rb") as f:
-    data = pickle.load(f)
+TOLERANCE = 0.4
+if not os.path.exists(SAVE_FILE):
```
linebreak between constants and initialisation section

HarrisonHG commented on main.py:
```diff
@@ -12,9 +13,13 @@
+else:
+    with open(SAVE_FILE, "rb") as f:
+        data = pickle.load(f)
 Knownencodings = data["encodings"]
```
Knownencodings -> known_encodings

underscores between words with all lower case is called **snake case** and is best for normal variables.

HarrisonHG commented on main.py:
```diff
@@ -25,19 +30,28 @@
-        matches = face_recognition.compare_faces(Knownencodings,encoding)
+    foundnames = []
+    for x,encoding in enumerate(encodings):
+        matches = face_recognition.compare_faces(Knownencodings,encoding, tolerance = TOLERANCE)
```
`tolerance = TOLERANCE` doesn't seem correct use here. Do you want to use TOLERANCE?

HarrisonHG commented on main.py:
```diff
@@ -25,19 +30,28 @@
+    print("Found:",foundnames)
     cv2.imshow("window",frame)
     if cv2.waitKey(1) == ord("q"):
         break
```
A good habit to get into at the end of loops like this that span the entire life of the program, is to add a 1 millisecond sleep so that the processor isn't burned out at potentially infinite speed.

HarrisonHG commented on headshots.py:
```diff
@@ -29,13 +36,11 @@
+    if k%256 == 32:
         # SPACE pressed
-        img_name = "dataset/"+ name +"/image_{}.jpg".format(img_counter)
+        img_name = DATASET_FOLDER+"/"+ NAME +"/image_{}.jpg".format(img_counter)
```
This could be better formatted as an f-string since oyu have multiple variables in a string.

eg - 

```
menu = ("eggs", "spam", 42.4)

f_string_order = f"{menu[0]} and {menu[1]}: {menu[2]:0.2f} ¤"
```

HarrisonHG replied:
pylint is insisting on treating this variable as a constant, so i recommend adding this comment to the end of the line: `# pylint: disable=invalid-name`

HarrisonHG commented on headshots.py:
```diff
@@ -1,19 +1,26 @@
-    os.makedirs(NAME)
+
 
 img_counter = 0
```
pylint is insisting on treating this variable as a constant, so i recommend adding this comment to the end of the line: `# pylint: disable=invalid-name`

HarrisonHG commented on main.py:
Now would be a good time to tidy up the console output from the program. Something less spammy would be nice, possibly hide the current output behind the debug logger introduced in my PR?

![image](https://github.com/user-attachments/assets/5b1085da-04f0-4dc0-bd42-0bf8089be599)

HarrisonHG replied:
![image](https://github.com/user-attachments/assets/d4f36d40-60ea-4cfc-8c1a-211487bc13c9)

HarrisonHG replied:
A simple "Found <name>, <name> and <name>" kind of message would be a pleasant sight.

HarrisonHG reviewed (changes requested):
Importantly, the code runs and works on my machine!

The comments are for linting and general cleaning up.

Jdavies08 committed:
changes for pylint standards and ease of use

changed everything to comply with pylint and reduced the number
of prints as well as allowing the user to input the name instead of
changing it manually

Jdavies08 commented:
Once this is approved i can resolve the conflicts

Jdavies08 committed:
Merge branch 'master' into test_feature
